src/java.desktop/share/classes/javax/swing/plaf/synth/SynthDesktopIconUI.java

```diff
@@ -108,10 +132,6 @@ protected void installListeners() {
      */
     @Override
     protected void uninstallListeners() {
-        if (iconPane instanceof JToggleButton) {
-            ((JToggleButton)iconPane).removeActionListener(handler);
-            frame.removePropertyChangeListener(this);
```

Comment:
This code is never executed, the iconPane is null here, it is moved above to the uninstallComponents()